persistence/src/main/scala/com/lightbend/lagom/internal/persistence/akkaserializer/OptionalSerializer.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.akkaserializer
+
+import java.util.Optional
+
+import akka.actor.ExtendedActorSystem
+import akka.serialization.{ BaseSerializer, SerializationExtension, Serializer, SerializerWithStringManifest }
+
+/**
+ * Base Serializer to support Optional<T> as a State or ReplyType on a PersistentEntity.
+ */
+class OptionalSerializer(val system: ExtendedActorSystem)
+  extends SerializerWithStringManifest with BaseSerializer {
+
+  private val separator = ':'
+
+  private val emptyManifest: String = "E"
+
+  // Must be lazy otherwise there's an infinite loop when loading the SerializationExtension
+  lazy val serialization = SerializationExtension(system)
+
+  def serializer(clazz: Class[_]): Serializer = serialization.serializerFor(clazz) // .asInstanceOf[SerializerWithStringManifest]
+
+  override def manifest(obj: AnyRef): String = {
+    val optional: Optional[AnyRef] = obj.asInstanceOf[Optional[AnyRef]]
+    if (optional.isPresent) {
+      val obj1 = optional.get()
+      val fqcn = obj1.getClass.getCanonicalName
+      val srlzr = serializer(obj1.getClass)
+
+      if (srlzr.includeManifest) {
+        val manifest = srlzr.asInstanceOf[SerializerWithStringManifest].manifest(obj1)
```

Comment:
This should perhaps be a `match` statement, and handle the case where the serializer is not a `SerializerWithStringManifest`.  I think in that case the manifest is simply a `Class`, and the class name should be written then.